lib/facility.rb

```diff
@@ -1,14 +1,85 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services,
+              :registered_vehicles,
+              :hours,
+              :holidays_closed,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_hash)
+    @name = facility_hash[:name]
+    @address = facility_hash[:address]
+    @phone = facility_hash[:phone]
+    @hours = facility_hash[:hours]
+    @holidays_closed = facility_hash[:holidaysclosed]
+    @collected_fees = 0
     @services = []
+    @registered_vehicles = []
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
-end
+
+  def register_vehicle(vehicle)
+    if @services.include?('Vehicle Registration')
+      registration_fee(vehicle)
+      vehicle.update_registration_date
+      vehicle.update_plate_type(vehicle)
+      update_registered_vehicles(vehicle)
+      
+    else
+      "Facility does not offer this service"
+    end
+  end
+
+  def update_registered_vehicles(vehicle)
+    @registered_vehicles << vehicle
+
+  end
+
+  def registration_fee(vehicle)
+    if vehicle.antique?
+      @collected_fees = 25
+    elsif vehicle.electric_vehicle?
+      @collected_fees = 200
+    else
+      @collected_fees = 100
+    end
+  end
+
+  def administer_written_test(registrant)
+    if @services.include?('Written Test')
+      registrant.license_data[:written] = true
+    else
+      "Facility does not offer this service"
+    end
+  end
+
+  def administer_road_test(registrant)
+    if @services.include?('Driving Test') 
+      if registrant.license_data[:written] == true
```

Comment:
This level of nested if statements is okay because the conditions are simple, and this allows you to give specific error messages. Just be wary of doing any *further* nesting or making the conditions more complex as that would probably get unreadable. But this looks great!